Updating Traffic Incidents

conor-ettinoffe-here opened the pull request:
- Add warning_level to be available
- Adding traffic_flow to show the effect of the incident has had on the flow of traffic

nvkelso commented:
Regarding `warning_level`, I don't think we should include this as it's **routing** focused not _map display_ focused. It should inform the `min_zoom`, and even a possible [collision](https://github.com/tilezen/vector-datasource/issues/988) rank. Please make a stronger case for it if you want it.

Previous discussion around this from Oct. 5th:

- https://github.com/tilezen/vector-datasource/issues/1598#issuecomment-425464977
- https://github.com/tilezen/vector-datasource/issues/1598#issuecomment-427267666
- https://github.com/tilezen/vector-datasource/issues/1598#issuecomment-427376549

I think we're already setting the **kind_detail** of `traffic_flow` features in the **traffic_flow** layer to those values. So again I'm not sure why this change is necessary.  Please provide more information.

- https://github.com/tilezen/vector-datasource/blob/ecf199a94c7ef1510040a2742041b51065fe1a4c/docs/layers.md#possible-kind-values-for-traffic-flow

conor-ettinoffe-here commented:
> Regarding `warning_level`, I don't think we should include this as it's **routing** focused not _map display_ focused. It should inform the `min_zoom`, and even a possible [collision](https://github.com/tilezen/vector-datasource/issues/988) rank. Please make a stronger case for it if you want it.
> 
> Previous discussion around this from Oct. 5th:
> 
> * [#1598 (comment)](https://github.com/tilezen/vector-datasource/issues/1598#issuecomment-425464977)
> * [#1598 (comment)](https://github.com/tilezen/vector-datasource/issues/1598#issuecomment-427267666)
> * [#1598 (comment)](https://github.com/tilezen/vector-datasource/issues/1598#issuecomment-427376549)
> 
> I think we're already setting the **kind_detail** of `traffic_flow` features in the **traffic_flow** layer to those values. So again I'm not sure why this change is necessary. Please provide more information.
> 
> * https://github.com/tilezen/vector-datasource/blob/ecf199a94c7ef1510040a2742041b51065fe1a4c/docs/layers.md#possible-kind-values-for-traffic-flow

Hi @nvkelso,

Within the discussion it was said "warning_level could be used to filter out incidents based on zoom level, so I think that using that for min_zoom would work fine". But one could also want to use the `warning_level` for other purposes such as displaying more dangerous incidents larger than other for instance.

Regarding the addition of `traffic_flow` it doesn't necessarily need to be of that name, but what I was trying to express with this field is the effect that the Incident has on the traffic around it, for example. There can be an accident, this could simply be slowing down the traffic flow because it has blocked one lane - or it could be blocking the whole road, thus freezing the flow of traffic.

nvkelso commented:
I can see adding `warning_level` as an **optional** property (not common, not common-optional). Really that's a design decision that's individual to a map style implementation and entirely dependent on veracity of that data (which you have well covered ;). Please update the PR.

But I continue to view `traffic_flow` property in this context as whole routing focused and inappropriate for map display.

conor-ettinoffe-here commented:
> I can see adding `warning_level` as an **optional** property (not common, not common-optional). Really that's a design decision that's individual to a map style implementation and entirely dependent on veracity of that data (which you have well covered ;). Please update the PR.

Moved the warning level to optional :)

> But I continue to view `traffic_flow` property in this context as whole routing focused and inappropriate for map display.

I've removed the field. Will need to see if there's another way to fulfil such a use case. I can see how you see this as routing focused, the use case one is trying to fulfill with including such information is to display the effect of the incident.
For example, a construction icon with a black background when its a fully disruptive construction site (No traffic), or a green background when its not obstructing traffic, or yellow when its causing some disruptions.

nvkelso commented:
> > But I continue to view `traffic_flow` property in this context as whole routing focused and inappropriate for map display.
> 
> I've removed the field. Will need to see if there's another way to fulfil such a use case. I can see how you see this as routing focused, the use case one is trying to fulfill with including such information is to display the effect of the incident.
> For example, a construction icon with a black background when its a fully disruptive construction site (No traffic), or a green background when its not obstructing traffic, or yellow when its causing some disruptions.

That sounds like a very technical display of data rather than for a general audience ;) 

My guess is the combination of `min_zoom` for these features in this layer, and also displaying with the **traffic_flow** lines from the other layer will mostly achieve what you want. But I'm happy to revisit in the next release. Let's get this merged now!
